intel_extension_for_transformers/llm/quantization/nn/modules.py

```diff
@@ -0,0 +1,186 @@
+                x, self.weight.data, out, self.out_features,
+                self.in_features, self.out_features, self.compute_dtype, self.weight_dtype
+            )
+
+        return out
+    
+    def set_weights_bias(self, weight_data, bias=None):
+        weight = torch.ops.weight_only_jblasop.qbits_quantize(
+            weight_data, True, self.blocksize, self.compute_dtype, self.weight_dtype)
+        self.weight = ParamsQBits(
+            data=weight, requires_grad=False, quant_state={"scheme": self.scheme}, blocksize=self.blocksize,
+            compress_statistics=self.compress_statistics, quant_dtype=self.weight_dtype
+        )
+        if bias is not None:
+            self.bias = torch.nn.Parameter(bias, requires_grad=False)
+
+
+class QuantizedLinearINT4(QuantizedLinearQBits, LoraLayer):
+    def __init__(
+        self,
+        input_features,
+        output_features,
+        bias=True,
+        compute_dtype="fp32",
+        compress_statistics=True,
+        weight_dtype="s4fullrange_scalef32",
+        blocksize=32,
+        scheme="sym",
+        device=None,
+        use_lora=True,
+        r=8,
+        lora_alpha=16,
+        lora_dropout=0.01,
+    ):
+        QuantizedLinearQBits.__init__(self, input_features, output_features, bias, compute_dtype, compress_statistics,
+                                      weight_dtype, blocksize, scheme, device)
+        self.use_lora = use_lora
+        if self.use_lora:
+            LoraLayer.__init__(self, input_features, output_features)
+            self._r = r
+            self._lora_alpha = lora_alpha
+            self._lora_dropout = lora_dropout
+
+    def forward(self, x: torch.Tensor):
+        if self.bias is not None and self.bias.dtype != x.dtype:
+            self.bias.data = self.bias.data.to(x.dtype)
+
+        # if getattr(self.weight, 'quant_state', None) is None:
+        #     print('FP4 quantization state not initialized. Please call .quantize_weights().')
+        inp_dtype = x.dtype
+        if self.compute_dtype is not None:
+            x = x.to(self.compute_dtype)
+
+        bias = None if self.bias is None else self.bias.to(self.compute_dtype)
+        out = matmul_4bit(x, self.weight, bias=bias, quant_state=self.weight.quant_state)
+        x = x.to(self.lora_A[self.active_adapter].weight.dtype)
+
+        if self.use_lora:
+            out += (
+                self.lora_B[self.active_adapter](
+                    self.lora_A[self.active_adapter](self.lora_dropout[self.active_adapter](x))
+                )
+                * self.scaling[self.active_adapter]
+            )
+
+        return out
+
+    def set_weights_bias(self, weight_data, bias=None):
+        self.weight = ParamsQBits(
+            weight_data, requires_grad=False, compress_statistics=self.compress_statistics, quant_dtype=self.weight_dtype
+        )
+        torch.ops.weight_only_jblasop.jblas_symqdq_weight(self.weight, False, 4, 32) # TODO: change to 4bit quantize instead of qdq
```

Comment:
have we added the uts for this pr? jblas_symqdq_weight has been replaced with jblas_dequantize now.

Reply:
we will add training part for this function， it will cover this.